sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/grouping.scala

```diff
@@ -144,12 +147,12 @@ case class GroupingSets(
 object GroupingSets {
   def apply(
       groupingSets: Seq[Seq[Expression]],
-      userGivenGroupByExprs: Seq[Expression]): GroupingSets = {
-    val groupingSetIndexes = GroupingSet.computeGroupingSetIndexes(groupingSets)
+      userGivenGroupByExprs: Seq[Expression]): GroupingAnalytic = {
+    val groupingSetIndexes = GroupingAnalytic.computeGroupingSetIndexes(groupingSets)
     GroupingSets(groupingSetIndexes, groupingSets.flatten, userGivenGroupByExprs)
   }
 
-  def apply(groupingSets: Seq[Seq[Expression]]): GroupingSets = {
+  def apply(groupingSets: Seq[Seq[Expression]]): GroupingAnalytic = {
```

Comment:
we can probably do the rename in a separate PR.

Reply:
> we can probably do the rename in a separate PR.

Done https://github.com/apache/spark/pull/32073